Added SpyFakku

KenjieDec opened the pull request:
Closes #3523 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

KenjieDec committed:
Added SpyFakku

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/Filters.kt:
```diff
@@ -0,0 +1,40 @@
+fun getFilters(): FilterList {
+    return FilterList(
+        SortFilter("Sort by", getSortsList),
+        OrderFilter("Order by", getOrdersList),
```
there is a `SortFilter` abstract clas which holds sort and ascending/descending state, use that to combine sort and order

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+
+    private val json: Json by injectLazy()
+
+    override val client = network.cloudflareClient
```
add some rate limit (2/s)

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+
+        val mangas = document.select("article.entry").map(::popularMangaFromElement)
+
+        val hasNextPage = document.select(".next").firstOrNull() != null
```
```suggestion
        val hasNextPage = document.selectFirst(".next") != null
```

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+
+    private fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        title = element.selectFirst("a")!!.attr("title")
```
```suggestion
        with(element.selectFirst("a")!!) {
            setUrlWithoutDomain(absUrl("href"))
            title = attr("title")
        }
```

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+
+                    is TextFilter -> {
+                        if (it.state.isNotEmpty()) {
+                            terms += it.state.split(",").map { tag ->
```
filter out empty

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+            }
+            append(
+                "Created At: ",
+                SimpleDateFormat("EEEE, d MMM yyyy HH:mm (z)", Locale.ENGLISH).format(
```
make this a class val, to avoide reinitializing it everytime

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+                .trim()
+                .replace(Regex("""^\?"""), "")
+                .lowercase()
+                .split(Regex("\\s+"))
```
move out these regex too

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakkuDto.kt:
```diff
@@ -0,0 +1,24 @@
+    private val name: String,
+) {
+    val value = name
+}
```
```suggestion
class Name(
    @SerialName("name") val value: String,
)
```

AwkwardPeak7 commented on src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:
```diff
@@ -0,0 +1,180 @@
+        author = artists?.joinToString { it.value }
+        artist = artists?.joinToString { it.value }
+        genre = tags?.joinToString { it.value }
+        thumbnail_url = "https://cdn.fakku.cc/data/$id/1/288.webp"
```
save cdnurl as a seperate val, so it is easier to change in the future

KenjieDec committed:
Changes

- Combined SortFilter and OrderFilter into One Filter.Sort 
- Remove useless regex
- Filter out empty tags
- Rate limit added 
- etc
